### PR TITLE
Add abstract and swappable models

### PR DESCRIPTION
**Pull request by slurms**
Just building on @auvipy's PR here.

Todo:
 - [x] Use the swappable models everywhere in the library.
 - [x] Update documentation.
 - [x] Test with swapped models.

### CODE REVIEW

**Comment by slurms** [comment]
@auvipy I'm just adding some custom model tests and more updated documentation, but this should be ready sometime soon :)

File: actstream/managers.py [hunk 1]
```diff
@@ -96,9 +91,9 @@ def user(self, obj, with_user_activity=False, **kwargs):
                 actor_object_id=obj.pk
             )
 
-        follows = apps.get_model('actstream', 'follow').objects.filter(user=obj)
+        follows = get_follow_model().objects.filter(user=obj)
         content_types = ContentType.objects.filter(
-            pk__in=Subquery(follows.values('content_type_id'))
```
Comment (auvipy) [inline actstream/managers.py]: why this is removed since it was added recently?

File: actstream/managers.py [hunk 2]
```diff
@@ -108,31 +103,23 @@ def user(self, obj, with_user_activity=False, **kwargs):
             object_ids = follows.filter(content_type=content_type)
             q = q | Q(
                 actor_content_type=content_type,
-                actor_object_id__in=Subquery(object_ids.values('object_id'))
```
Comment (auvipy) [inline actstream/managers.py]: same as above

**Review by auvipy — COMMENTED** [review]
are you going to revisit this anytime soon?

**Pull request closed without merging**